source-code/core2/src/messages/query.ts

```diff
@@ -1,15 +1,31 @@
 import type { Message } from "./schema.js"
 
-// TODO
-// 1. Query builder?
-// 2. Filtering etc how?
-//
+type UniqueFilterType = {
+	id: string
+	languageTag: string
+	//variant: string -> in the future
+}
+type FilterType = {
+	id?: string
+	languageTag?: string
+	//variant?: string -> in the future
+}
+
 // inspiration: Prisma CRUD https://www.prisma.io/docs/concepts/components/prisma-client/crud
 export type MessagesQueryApi_1 = {
-	create: (message: Message) => void
-	get: () => Message[]
-	update: () => void
-	delete: () => void
+	create: (message: Message) => Message
+	get: (where: UniqueFilterType) => Message
+	getMany: (where: FilterType) => Message[]
+	update: (where: UniqueFilterType, data: Message["pattern"]) => Message
```

Comment:
update should also be possible with a single payload like this:
```ts
const message = inlang.messages.query.update({ id: "myMessageId", languageTag: "en", pattern: [{"type": "Text", value: "Hello World" }] })
```

Reply:
Should both be possible? I feel like we should decide on one.